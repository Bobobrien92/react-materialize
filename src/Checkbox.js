import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import idgen from './idgen';

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: props.checked
    };

    this.id = props.id || idgen();
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const { onChange } = this.props;
    if (onChange) onChange(event);

    this.setState({ checked: event.target.checked });
  }

  render() {
    const { filledIn, label, disabled, value } = this.props;
    const checkboxClassName = { 'filled-in': filledIn };

    return (
      <label>
        <input
          className={cx(checkboxClassName)}
          disabled={disabled}
          onChange={this.handleChange}
          type="checkbox"
          checked={this.state.checked}
          value={value}
        />
        <span>{label}</span>
      </label>
    );
  }
}

Checkbox.propTypes = {
  /*
   * checkbox value
   */
  value: PropTypes.string.isRequired,
  /*
   * filled-in styled checkbox
   */
  filledIn: PropTypes.bool,
  /*
   * label next to checkbox
   */
  label: PropTypes.string.isRequired,
  /*
   * onChange callback
   */
  onChange: PropTypes.func,
  /*
   * override id
   * @default idgen()
   */
  id: PropTypes.string,
  /*
   * disabled input
   */
  disabled: PropTypes.bool,
  /*
   * initialise checkbox checked
   */
  checked: PropTypes.bool
};

export default Checkbox;
