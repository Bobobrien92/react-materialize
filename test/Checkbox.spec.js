import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../src/Checkbox';

describe('<Checkbox />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(<Checkbox value="red" label="red" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders disabled', () => {
    wrapper = shallow(<Checkbox value="red" label="red" disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  test('filled-in', () => {
    wrapper = shallow(<Checkbox value="red" label="red" filledIn />);
    expect(wrapper).toMatchSnapshot();
  });

  test('handles checked state', () => {
    wrapper = shallow(<Checkbox value="red" label="red" checked />);
    expect(wrapper.state().checked).toEqual(true);
  });
});
