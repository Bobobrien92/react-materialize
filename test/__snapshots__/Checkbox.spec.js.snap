// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> filled-in 1`] = `
<label
  htmlFor={2}
>
  <input
    className="filled-in"
    id={2}
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> renders 1`] = `
<label
  htmlFor={0}
>
  <input
    className=""
    id={0}
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> renders disabled 1`] = `
<label
  htmlFor={1}
>
  <input
    className=""
    disabled={true}
    id={1}
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;
